rust/operator/src/kafka_controller.rs

```diff
@@ -894,12 +924,14 @@ fn build_broker_rolegroup_statefulset(
             }),
             ..Volume::default()
         })
-        .security_context(PodSecurityContext {
-            run_as_user: Some(1000),
-            run_as_group: Some(1000),
-            fs_group: Some(1000),
-            ..PodSecurityContext::default()
-        });
+        .service_account_name(sa_name)
+        .security_context(
+            PodSecurityContextBuilder::new()
+                .run_as_user(KAFKA_UID)
+                .run_as_group(0)
+                .fs_group(1000) // Needed for secret-operator
```

Comment:
There are other reasons as well
```suggestion
                .fs_group(1000)
```